### ISSUE TITLE
[bug]Typecheck does not recognize alias assign

### ISSUE THREAD

**Issue by SolaWing** [castwide/solargraph#497]
```
class UserInterface
  def self.info(); end
end

UI = UserInterface

# @return [UI]
def get # <-- Unresolved return type UI for #get
  UI
end

get.info
```

**Comment by castwide** [comment]
Confirmed. There are existing mechanisms to recognize constants that reference classes in this manner, so it should be fixable.

**Comment by castwide** [comment]
Released in v0.44.3.

**Comment by castwide** [comment]
Closing as resolved.
